moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -799,6 +799,45 @@ bool moveit::core::RobotState::isValidVelocityMove(const RobotState& other, cons
         return false;
     }
   }
+  return true;
+}
+
+bool moveit::core::RobotState::isValidVelocityMove(const JointModelGroup* group,
+                                                   const std::vector<double>& from_joint_pose,
+                                                   const std::vector<double>& to_joint_pose, double dt) const
+{
+  // Check for equal sized arrays
+  if (from_joint_pose.size() != to_joint_pose.size())
+  {
+    logError("To and from joint poses are of different sizes.");
+    return false;
+  }
+
+  return isValidVelocityMove(group, &from_joint_pose[0], &to_joint_pose[0], from_joint_pose.size(), dt);
+}
+
+bool moveit::core::RobotState::isValidVelocityMove(const JointModelGroup* group, const double* from_joint_pose,
+                                                   const double* to_joint_pose, std::size_t array_size, double dt) const
+{
+  const std::vector<const JointModel::Bounds*>& bounds = group->getActiveJointModelsBounds();
+  const std::vector<unsigned int>& bij = group->getKinematicsSolverJointBijection();
+
+  for (std::size_t i = 0; i < array_size; ++i)
+  {
+    double dtheta = std::abs(from_joint_pose[i] - to_joint_pose[i]);
+    const std::vector<moveit::core::VariableBounds>* var_bounds = bounds[bij[i]];
+
+    if (var_bounds->size() != 1)
+    {
+      logError("Attempting to check velocity bounds for waypoint move with joints that have multiple variables");
```

Comment:
I'm not sure, how to handle non-standard joints here. Why do you want to bail out? You could also skip those joints. 
Now, it's not possible to use this function for JMG's containing non-standard joints.

Reply:
This is a future todo. I'm not inclined to support it since its an uncommon use case, but when the need arises i welcome the change

I added a TODO as such